app/workflow_executors/step_executors/impl/next_js_server_test_executor.go

```diff
@@ -170,46 +173,68 @@ func (e NextJsServerStartTestExecutor) Execute(step steps.ServerStartTestStep) e
-	}
-	claudeClient:= llms.NewClaudeClient(apiKey)
-	response, err := claudeClient.ChatCompletion(messages)
-	if err != nil {
-		settingsUrl := config.Get("app.url").(string) + "/settings"
-		err = e.activityLogService.CreateActivityLog(
-			step.Execution.ID,
-			step.ExecutionStep.ID,
-			"INFO",
-			fmt.Sprintf("Action required: There's an issue with your LLM API Key. Ensure your API Key for %s is correct. <a href='%s' style='color:%s; text-decoration:%s;'>Settings</a>", constants.CLAUDE_3, settingsUrl, "blue", "underline"),
-		)
-		if err != nil {
-			fmt.Printf("Error creating activity log: %s\n", err.Error())
-			return false, nil, err
+	claudeClient := llms.NewClaudeClient(apiKey)
+	var jsonResponse map[string]interface{}
+	var response string
+
+	for retryCount := 1; retryCount < 6; retryCount++ {
+		messages, err := e.CreateMessage(buildLogs, directoryPlan, retryCount)
+		if err != nil{
+			fmt.Println("failed to create messages for llm")
+            return false, nil, err
 		}
-		//Update Execution Status and Story Status
-		if err = e.storyService.UpdateStoryStatus(int(step.Story.ID), constants.InReviewLLMKeyNotFound); err != nil {
-			fmt.Printf("Error updating story status: %s\n", err.Error())
-			return false, nil, err
+		response, err = claudeClient.ChatCompletion(messages)
+		if err != nil {
+			settingsUrl := config.Get("app.url").(string) + "/settings"
+			err = e.activityLogService.CreateActivityLog(
+				step.Execution.ID,
+				step.ExecutionStep.ID,
+				"INFO",
+				fmt.Sprintf("Action required: There's an issue with your LLM API Key. Ensure your API Key for %s is correct. <a href='%s' style='color:%s; text-decoration:%s;'>Settings</a>", constants.CLAUDE_3, settingsUrl, "blue", "underline"),
+			)
+			if err != nil {
+				fmt.Printf("Error creating activity log: %s\n", err.Error())
+				return false, nil, err
+			}
+			//Update Execution Status and Story Status
+			if err = e.storyService.UpdateStoryStatus(int(step.Story.ID), constants.InReviewLLMKeyNotFound); err != nil {
+				fmt.Printf("Error updating story status: %s\n", err.Error())
+				return false, nil, err
+			}
+			if err = e.executionService.UpdateExecutionStatus(step.Execution.ID, constants.InReviewLLMKeyNotFound); err != nil {
+				fmt.Printf("Error updating execution step: %s\n", err.Error())
+				return false, nil, err
+			}
+			fmt.Println("failed to generate code from llm")
+			if retryCount == 4 {
+				return false, nil, fmt.Errorf("failed to generate code from llm after 5 attempts: %w", err)
+			}
+			continue
 		}
-		if err = e.executionService.UpdateExecutionStatus(step.Execution.ID, constants.InReviewLLMKeyNotFound); err != nil {
-			fmt.Printf("Error updating execution step: %s\n", err.Error())
-			return false, nil, err
+		if err = json.Unmarshal([]byte(response), &jsonResponse); err != nil {
+			fmt.Println("failed to unmarshal response from Claude API, retrying...")
+			if retryCount == 5 {
+				err := e.slackAlert.SendAlert(
```

Comment:
put error in each case even when ther is single retry may be message can be different that will help us figure out how many time such cases happen